scripts/dashboard/validate_dashboards_format.py

```diff
@@ -0,0 +1,153 @@
+    dashDict = get_sample_dashboards_json(jsonPath)
+
+    if not 'displayName' in dashDict.keys():
+      raise JsonFormattingError("{} is missing displayName field".format(jsonFile))
+
+    pathMap[directory]['jsonNames'][dashDict['displayName']]=''
+
+    fileNameParts = jsonFile.split('.')
+    check_json_file_name(jsonFile, fileNameParts)
+
+    pathMap[directory]['jsonIds'][fileNameParts[0]]=''
+
+def check_metadata_entries(path, sample_dashboard):
+  '''Check if metadata file fields are as expected'''
+  required_fields = {"category", "id", "display_name", "description"}
+  missing_fields = required_fields - sample_dashboard.keys()
+  if missing_fields:
+    raise MetadataFormattingError("{} missing {}".format(path, missing_fields))
+
+def check_metadata(directory, pathMap):
+  '''Checks metadata file in directory for correct fields and matching data in 
+  json files within the same directory'''
+  path = os.path.join('.', 'dashboards', directory, 'metadata.yaml')
+
+  with open(path) as f:
+    data = yaml.safe_load(f)
+  sample_dashboards = data.get("sample_dashboards")
+  if not sample_dashboards:
+    raise MetadataFormattingError("sample_dashboards not defined in {}".format(path))
+  for sample_dashboard in sample_dashboards:
+    # Call function to check metadata file fields
+    check_metadata_entries(path, sample_dashboard)
+
+    dashboardId = sample_dashboard.get('id')
+    dashboardName = sample_dashboard.get('display_name')
+
+    pathMap[directory]['metadataIds'][dashboardId]=''
+    pathMap[directory]['metadataNames'][dashboardName]=''
+
+    if not dashboardId in pathMap[directory]['jsonIds'].keys():
+      raise MetadataInJsonMisMatchError("{} does not have a matching json file".format(dashboardId))
+    
+    if not dashboardName in pathMap[directory]['jsonNames'].keys():
+      raise MetadataInJsonMisMatchError("{} does not have a matching json file".format(dashboardName))
+
+def check_json_in_metadata(directory, pathMap):
+  '''Checks json file data against metadata file of the same directory'''
+  for jsonFile in pathMap[directory]['jsonFiles']:
+    jsonPath = os.path.join('.', 'dashboards', directory, jsonFile)
+    dashDict = get_sample_dashboards_json(jsonPath)
+
+    if not jsonFile.split('.')[0] in pathMap[directory]['metadataIds'].keys():
+      raise JsonInMetadataMisMatchError("{} does not have a matching id in the metadata file".format(jsonFile))
+    
+    if not dashDict['displayName'] in pathMap[directory]['metadataNames'].keys():
+      raise JsonInMetadataMisMatchError("{} does not have a matching display_name in the metadata file".format(jsonFile))
+
+def check_directory(directory, pathMap):
+  '''Parent function that calls top level processing and checking functions'''
+
+  # Process json files into path map
+  mapJsonFiles(directory, pathMap)
+
+  # Metadata file check
+  check_metadata(directory, pathMap)
+
+  # json file check
+  check_json_in_metadata(directory, pathMap)
```

Comment:
It would be worth adding to these comments to call out that they are order dependent, since they are mutating `pathMap`.

Reply:
done in docstring for parent function.